crates/core/src/dht/dht_store.rs

```diff
@@ -51,14 +46,8 @@ pub struct DhtStore {
 
 impl PartialEq for DhtStore {
     fn eq(&self, other: &DhtStore) -> bool {
-        let content = &self.content_storage.clone();
-        let other_content = &other.content_storage.clone();
-        let meta = &self.meta_storage.clone();
-        let other_meta = &other.meta_storage.clone();
-
         self.holding_map == other.holding_map
-            && (*content.read().unwrap()).get_id() == (*other_content.read().unwrap()).get_id()
-            && *meta.read().unwrap() == *other_meta.read().unwrap()
+            && (*self.persistence_manager).get_id() == (*other.persistence_manager).get_id()
```

Comment:
I saw another Eq operation like this with the derefs / get_id() thing - should this be a PartialEq/Eq on the persistence_manager itself?

Reply:
Agreed that this could be a partialEQ on the Persistence Manager itself